### PR TITLE
Added an option to disable processing of @ApplicationPath.

### PR DESCRIPTION
**Pull request by EricWittmann**

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java [hunk 1]
```diff
@@ -350,7 +350,7 @@ private OpenAPIImpl jaxRsApplicationToOpenApi(ClassInfo applicationClass) {
         ////////////////////////////////////////
         AnnotationInstance appPathAnno = JandexUtil.getClassAnnotation(applicationClass,
                 OpenApiConstants.DOTNAME_APPLICATION_PATH);
-        if (appPathAnno == null) {
+        if (appPathAnno == null || config.applicationPathDisable()) {
             appPathAnno = JandexUtil.getClassAnnotation(applicationClass, OpenApiConstants.DOTNAME_PATH);
         }
```
Comment (MikeEdgar) [inline implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java]: Is the intent that `@Path` on an `Application` instance would not be impacted by the new configuration?

**Comment by MikeEdgar** [comment]
@EricWittmann - this looks good. I'll wait for your feedback on the one comment before closing the review. I noticed that the two new test classes have 2018 as the copyright year. I'm not sure if that's important so I just wanted to mention it.

File: implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java (hunk 1 above)
Comment (EricWittmann) [reply implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java]: Yeah I wasn't sure about that.  But that's the decision I ended up going with.  What do you think?

**Comment by EricWittmann** [comment]
I fixed the copyright issues.  Damn you copy/paste. :)

File: implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java (hunk 1 above)
Comment (MikeEdgar) [reply implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java]: I think that's good. It keeps the configuration name in alignment with the annotation.

**Pull request merged**